pytorch_lightning/trainer/trainer.py

```diff
@@ -2471,6 +2474,39 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Union[LightningLoggerBase, None]) -> None:
+        if new_logger:
```

Comment:
This implementation could be just `self.loggers = new_logger if new_logger is None else [new_logger]`

Reply:
Very interesting, didn't occur to me to use one setter inside another

Reply:
I don't think this would unpack a `LoggerCollection` properly because it would assign `self.loggers = [LoggerCollection]`, so there would be an iterable inside an iterable. I like this idea though, so I'll incorporate it :)

Reply:
Something like:
`self.loggers = [new_logger] if new_logger and not isinstance(new_logger, LoggerCollection) else new_logger`